### PR TITLE
feat: end to end entry point routing

### PR DESCRIPTION
**Pull request by dancoombs**
## Proposed Changes

  - Implement RPC events trait for v0.7 (most code shared)
  - Wire up entry point 0.7 in various places, only supports `UNSAFE` and a hardcoded gas estimator
  - WIP: needs to be cleaned up and tested

### CODE REVIEW

File: crates/rpc/src/eth/router.rs [hunk 1]
```diff
@@ -60,6 +60,7 @@ impl EntryPointRouterBuilder {
                 route.version()
             );
         }
+        tracing::info!("Adding route for v0.7 entry point {:?}", route.address());
```
Comment (dancoombs) [inline crates/rpc/src/eth/router.rs:63]: TODO remove

File: crates/builder/src/task.rs [hunk 2]
```diff
@@ -255,6 +250,92 @@ where
         Box::new(self)
     }
 
+    async fn create_builders_v0_6<C, E>(
```
Comment (0xfourzerofour) [inline crates/builder/src/task.rs:253]: We should add a todo here to merge these builder creator functions into one using generics based on the UO version once entrypoint v7 safe simulation is implemented

File: crates/pool/src/task.rs [hunk 3]
```diff
@@ -182,18 +184,32 @@ impl PoolTask {
...
+                ep,
+                simulator,
+            )
+        } else {
+            panic!("V0_7 safe simulation not supported")
+        }
     }
 
-    async fn create_mempool_v0_6<P: Provider + Middleware>(
+    fn create_mempool_v0_6<P: Provider + Middleware>(
```
Comment (0xfourzerofour) [inline crates/pool/src/task.rs:212]: Same thing here, it looks possible but need to play around with the code for this feature to make sure

File: crates/sim/src/simulation/mempool.rs [hunk 4]
```diff
@@ -25,10 +25,20 @@ use crate::simulation::SimulationViolation;
 /// Typically read from a JSON file using the `Deserialize` trait.
 #[derive(Debug, Clone, Deserialize, Default)]
 pub struct MempoolConfig {
+    /// Entry point address this mempool is associated with.
+    #[serde(rename = "camelCase")]
+    pub(crate) entry_point: Address,
```
Comment (0xfourzerofour) [inline crates/sim/src/simulation/mempool.rs:30]: Will we have to add multiple mempool configs per network based on entrypoint addresses now?

**Review by 0xfourzerofour — APPROVED** [review]
Looks good, just added a couple comments to consolidate the code at a later date based on implementation

File: crates/builder/src/task.rs (hunk 2 above)
Comment (dancoombs) [reply crates/builder/src/task.rs:253]: Yeah I found the types a little difficult to work with to get a common function. Adding a TODO

File: crates/sim/src/simulation/mempool.rs (hunk 4 above)
Comment (dancoombs) [reply crates/sim/src/simulation/mempool.rs:30]: Added a TODO

File: crates/builder/src/task.rs [hunk 5]
```diff
@@ -143,45 +147,36 @@ where
...
-                        ),
-                    )
-                    .await?
-                };
-                sender_handles.push(spawn_guard);
-                bundle_sender_actions.push(bundle_sender_action);
+            match ep.version {
+                EntryPointVersion::V0_6 => {
+                    let (handles, actions) = self
+                        .create_builders_v0_6(ep, Arc::clone(&provider), ep_v0_6.clone())
```
[the remark this answers is not in the document]
Comment (dancoombs) [reply crates/builder/src/task.rs:165]: From an interface perspective I don't like this. Its possible in the future an entry point wouldn't have an inner provider, that is specific to the ethers implementation.

**Pull request merged**